pkg/migrate/migrate.go

```diff
@@ -245,26 +392,27 @@ func copyOnePVC(ctx context.Context, w *log.Logger, clientset k8sclient.Interfac
 		}
 	}()
 
-	// wait for the pod to be created
-	time.Sleep(waitTime)
+	migrationTimeout := time.NewTicker(timeout)
+	waitInterval := time.NewTicker(1 * time.Second)
+	defer migrationTimeout.Stop()
+	defer waitInterval.Stop()
 	for {
 		gotPod, err := clientset.CoreV1().Pods(ns).Get(ctx, createdPod.Name, metav1.GetOptions{})
 		if err != nil {
 			w.Printf("failed to get newly created migration pod %s: %v\n", createdPod.Name, err)
-			continue
-		}
-
-		if gotPod.Status.Phase == corev1.PodPending {
-			time.Sleep(waitTime)
-			continue
-		}
-
-		if gotPod.Status.Phase == corev1.PodRunning || gotPod.Status.Phase == corev1.PodSucceeded {
+		} else if gotPod.Status.Phase == corev1.PodRunning || gotPod.Status.Phase == corev1.PodSucceeded {
 			// time to get logs
 			break
 		}
 
 		w.Printf("got status %s for pod %s, this is likely an error\n", gotPod.Status.Phase, gotPod.Name)
+
+		select {
+		case <-waitInterval.C:
+			continue
+		case <-migrationTimeout.C:
+			return fmt.Errorf("migration pod %s failed to go into Running phase: timedout", createdPod.Name)
```

Comment:
have we a CR that represents the service? if so, could we use use conditional status? 
So that, we can properly set the status of this one?

Reply:
We don't have a CR. This project is both a CLI and library for doing PVC migrations. 
What do you mean by 
>if so, could we use use conditional status?
So that, we can properly set the status of this one?